src/main/java/net/dv8tion/jda/core/entities/impl/InviteImpl.java

```diff
@@ -0,0 +1,322 @@
+        this.inviter = inviter;
+        this.maxAge = maxAge;
+        this.maxUses = maxUses;
+        this.temporary = temporary;
+        this.timeCreated = timeCreated;
+        this.uses = uses;
+        this.channel = channel;
+        this.guild = guild;
+    }
+
+    public static RestAction<Invite> resolve(final JDA api, final String code)
+    {
+        Args.notNull(code, "code");
+        Args.notNull(api, "api");
+
+        final Route.CompiledRoute route = Route.Invites.GET_INVITE.compile(code);
+
+        return new RestAction<Invite>(api, route, null)
+        {
+            @Override
+            protected void handleResponse(final Response response, final Request request)
+            {
+                if (response.isOk())
+                {
+                    final Invite invite = EntityBuilder.get(this.api).createInvite(response.getObject());
+                    request.onSuccess(invite);
+                }
+                else
+                {
+                    request.onFailure(response);
+                }
+            }
+        };
+    }
+
+    @Override
+    public RestAction<Invite> delete()
+    {
+        final Route.CompiledRoute route = Route.Invites.DELETE_INVITE.compile(this.code);
+
+        return new RestAction<Invite>(this.api, route, null)
+        {
+            @Override
+            protected void handleResponse(final Response response, final Request request)
+            {
+                if (response.isOk())
+                {
+                    final Invite invite = EntityBuilder.get(this.api).createInvite(response.getObject());
+                    request.onSuccess(invite);
+                }
+                else
+                {
+                    request.onFailure(response);
+                }
+            }
+        };
+    }
+
+    @Override
+    public RestAction<Invite> expand()
+    {
+        if (this.expanded)
+            return new RestAction.EmptyRestAction<>(this);
+
+        final net.dv8tion.jda.core.entities.Guild guild = this.api.getGuildById(this.guild.getId());
+
+        if (guild == null)
+            throw new UnsupportedOperationException("You're not in the guild this invite points to");
+
+        final Member member = guild.getSelfMember();
+
+        CompiledRoute route;
+
+        final net.dv8tion.jda.core.entities.Channel channel = this.channel.getType() == ChannelType.TEXT
+                ? guild.getTextChannelById(this.channel.getId()) : guild.getVoiceChannelById(this.channel.getId());
+
+        if (member.hasPermission(channel, Permission.MANAGE_CHANNEL))
+        {
+            route = Route.Invites.GET_CHANNEL_INVITES.compile(channel.getId());
+        }
+        else if (member.hasPermission(Permission.MANAGE_SERVER))
+        {
+            route = Route.Invites.GET_GUILD_INVITES.compile(guild.getId());
+        }
+        else
+            throw new PermissionException("You don't have the permission to view the full invite info");
```

Comment:
Perhaps github is out of date, but it appears that the `else` -> `throw` still does not use `{ }`